newIDE/app/src/UI/Search/UseSearchItem.js

```diff
@@ -4,18 +4,39 @@ import { type ChosenCategory } from './FiltersChooser';
 import shuffle from 'lodash/shuffle';
 import SearchApi from 'js-worker-search';
 
+export interface SearchFilter<SearchItem> {
+  isSatisfiedBy(searchItem: SearchItem): boolean;
+}
+
+export class TagSearchFilter<SearchItem: { tags: Array<string> }>
```

Comment:
Is this class really needed? It seems to only have one subclass?

Reply:
It's to keep the logic about filtering on tags in this file. It might help to refactor when the old filters will be dropped or merged with the new filters.